File: backend/ControleGastos.API/Models/Person.cs
namespace ControleGastos.API.Models
{
    public class Person
    {
        public Guid Id { get; private set; }
        public string Name { get; private set; } = string.Empty;
        public DateOnly BirthDate { get; private set; }

        public ICollection<Transaction> Transactions { get; private set; } = [];

        private Person() { }

        public Person(string name, DateOnly birthDate)
        {
            Id = Guid.NewGuid();
            Name = name;
            BirthDate = birthDate;
        }

        public int GetAge()
        {
            var today = DateOnly.FromDateTime(DateTime.Today);
            var age = today.Year - BirthDate.Year;

            if (BirthDate > today.AddYears(-age))
            {
                age--;
            }

            return age;
        }
    }
}

File: frontend/controle-gastos-web/src/types/person.ts
export interface Person {
  id: string;
  name: string;
  age: number;
}

export interface CreatePerson {
  name: string;
  birthDate: string;
}

export interface PersonSummary {
  id: string;
  name: string;
  totalIncome: number;
  totalExpense: number;
  balance: number;
}